ospackage/recipes/steps.go

```diff
@@ -17,65 +17,45 @@ package recipes
 import (
 	"fmt"
 	"os"
+	"os/exec"
+	"path/filepath"
 	"strings"
 
 	osconfigpb "github.com/GoogleCloudPlatform/osconfig/_internal/gapi-cloud-osconfig-go/google.golang.org/genproto/googleapis/cloud/osconfig/v1alpha2"
 )
 
-// BuildCommand builds a command []string based on the Step type and parameters.
-func BuildCommand(step *osconfigpb.SoftwareRecipe_Step, artifacts map[string]string) ([]string, error) {
-	switch v := step.Step.(type) {
-	case *osconfigpb.SoftwareRecipe_Step_FileCopy:
-		return StepFileCopy(v, artifacts)
-	case *osconfigpb.SoftwareRecipe_Step_ArchiveExtraction:
-		return StepArchiveExtraction(v, artifacts)
-	case *osconfigpb.SoftwareRecipe_Step_MsiInstallation:
-		return StepMsiInstallation(v, artifacts)
-	case *osconfigpb.SoftwareRecipe_Step_DpkgInstallation:
-		return StepDpkgInstallation(v, artifacts)
-	case *osconfigpb.SoftwareRecipe_Step_RpmInstallation:
-		return StepRpmInstallation(v, artifacts)
-	case *osconfigpb.SoftwareRecipe_Step_FileExec:
-		return StepFileExec(v, artifacts)
-	case *osconfigpb.SoftwareRecipe_Step_ScriptRun:
-		return StepScriptRun(v, artifacts)
-	default:
-		return nil, fmt.Errorf("unknown step type %T", v)
-	}
-}
-
 // StepFileCopy builds the command for a FileCopy step
-func StepFileCopy(step *osconfigpb.SoftwareRecipe_Step_FileCopy, artifacts map[string]string) ([]string, error) {
+func StepFileCopy(step *osconfigpb.SoftwareRecipe_Step_FileCopy, artifacts map[string]string) error {
```

Comment:
where did filecopy go..?

Reply:
That's in a separate PR. I figured this refactor and the FileCopy are fairly different chunks of work.

Reply:
Currently they are parallel branches. My plan was to rebase the filecopy on top of this if it get's approved.